cocoa/src/toga_cocoa/window.py

```diff
@@ -43,12 +46,62 @@ def windowShouldClose_(self, notification) -> bool:
         self.interface.on_close()
         return False
 
+    @objc_method
+    def windowWillClose_(self, notification) -> None:
+        # Setting the toolbar delegate to None doesn't disconnect
+        # the delegate and still triggers the delegate events.
+        # Hence, simply remove the toolbar instead.
+        if self.toolbar:
+            self.toolbar = None
+        # Disconnect the window delegate.
+        self.delegate = None
+
+        # Disconnecting the window delegate doesn't prevent custom
+        # methods which are performed with a delay i.e., having a
+        # delay != 0), from being triggered. Hence, check guards
+        # needs to be used in such custom methods.
+
     @objc_method
     def windowDidResize_(self, notification) -> None:
         if self.interface.content:
             # Set the window to the new size
             self.interface.content.refresh()
 
+    @objc_method
+    def windowDidMiniaturize_(self, notification) -> None:
+        if (
+            self.impl._pending_state_transition
+            and self.impl._pending_state_transition != WindowState.MINIMIZED
+        ):
+            self.impl._apply_state(WindowState.NORMAL)
```

Comment:
No matter, how many different ways I rearrange the intermediate states, or change the delay timing, this line never gets covered under CI. But it gets covered on my local machine.

Reply:
I can confirmatory say that the line doesn't get covered in CI, but gets covered on my local machine. I had added a test to check it:
https://github.com/beeware/toga/blob/116eec3840740d091baa325928b2b9f49cf8073e/testbed/tests/window/test_window.py#L944-L963
This test covers line 76 on my local machine, but doesn't get covered in CI. This indicates that the OS delay timing for completing the state change operation is different on CI. So, I'll mark it as no-cover and add a comment explaining it.

Reply:
Yes, this is no longer an issue.